packages/expo-updates/src/Updates.ts

```diff
@@ -132,7 +132,7 @@ export async function reloadAsync(): Promise<void> {
   if (!ExpoUpdates.reload) {
     throw new UnavailabilityError('Updates', 'reloadAsync');
   }
-  if (__DEV__ && !isUsingExpoDevelopmentClient) {
+  if (!ExpoUpdates?.nativeDebug && (__DEV__ || isUsingExpoDevelopmentClient)) {
```

Comment:
@douglowder, before this change, if `isUsingExpoDevelopmentClient` was true, then the exception would not be thrown and the `reloadAsync()` call would succeed.

After the change, the exception will be thrown if `isUsingExpoDevelopmentClient` is true, unless `nativeDebug` is true.

Was this intended?

There are users who were using `reloadAsync()` for restarting the app for purposes other than OTA updates (e.g. after changing language settings). After this change, the dev client can't be restarted, unless `nativeDebug` is enabled.

https://forums.expo.dev/t/i-cant-restart-the-app-using-expo-updates-to-change-language/74237?u=wodin

Reply:
Good catch! I'll get this fixed.

Reply:
Thanks! I've posted a followup to the thread on the forums.

Reply:
@douglowder
It is working very well with me thanks to @wodin.

Reply:
PR is up at https://github.com/expo/expo/pull/24588